ratpack-core/src/main/java/ratpack/http/client/HttpClient.java

```diff
@@ -106,8 +112,20 @@ static HttpClient httpClient(ServerConfig serverConfig, Registry registry) {
    * @param maxContentLengthBytes The max content length of a response to support.
    * @return An instance of a HttpClient
    */
-  static HttpClient httpClient(ByteBufAllocator byteBufAllocator, int maxContentLengthBytes) {
-    return new DefaultHttpClient(byteBufAllocator, maxContentLengthBytes);
+  static HttpClient httpClient(ByteBufAllocator byteBufAllocator, int maxContentLengthBytes, ExecController execController) {
```

Comment:
This is a breaking binary change. You'll need to introduce this as a new method and then modify the existing method with 2 args to call the new method with a sane default for the new parameter.